lib/algorithms/rsa.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+const crypto = require('crypto');
+const { promisify } = require('util');
+
+const algorithms = require('../algorithms');
+const {
+  InvalidAccessError,
+  NotSupportedError
+} = require('../errors');
+const { kKeyMaterial, CryptoKey } = require('../key');
+const { limitUsages, opensslHashFunctionName, toBuffer } = require('../util');
+
+const generateKeyPair = promisify(crypto.generateKeyPair);
+
+function uint8ArrayToUint32(bigInteger) {
+  let result = 0;
+  for (let i = 0; i < bigInteger.length; i++)
+    result = (result << 8) | bigInteger[i];
+  return result;
+}
+
+function uint32ToUint8Array(integer) {
+  const result = Buffer.alloc(8);
+  let i = 7;
+  while (integer !== 0) {
+    result[i--] = integer & 0xff;
+    integer >>= 8;
+  }
+  return result.slice(i + 1);
```

Comment:
returns a Buffer, not a Uint8Array, does that matter?

Reply:
I don't think it matters since every `Buffer` is also an `Uint8Array` (so `instanceof Uint8Array` would return true).